File: prob36/prob36.js
function countZeros(array) {
	var count = 0;
	for (var i = 0; i < array.length; i++) {
		array[i] === 0 ? count++ : null;
	}
	return count;
}

console.log(countZeros([0,1,0,2,3]))
console.log(countZeros([0,1,0,0,0]))
console.log(countZeros([0]))


//** refactor with a helper function */

function countZerosHelper(array) {
	var count = 0;
	for (var i = 0; i < array.length; i++) {
		hasZeros(array[i]) === true ? count++ : null;
	}
	return count;
}

function hasZeros(integer){
	integer === 0 ? true : false;
}

console.log(countZerosHelper([0,1,0,2,3]))
console.log(countZerosHelper([0,1,0,0,0]))
console.log(countZerosHelper([0]))

File: problems/prob36/prob36.rb
def count_zeros(array)
	count = 0
	array.each { |int| int == 0 ? count +=1 : nil }
	return count
end

puts count_zeros([0,1,0,2,3]) == 2
puts count_zeros([0,1,0,0,0]) == 4
puts count_zeros([0]) == 1

### -- refactured with a helper function -- ###

def count_zeros_with_helpers(array)
	count = 0
	array.each { |int| has_zeros(int) == true ? count +=1 : nil}
	return count
end

def has_zeros(num)
	num == 0 ? true : false
end

puts count_zeros_with_helpers([0,1,0,2,3]) == 2
puts count_zeros_with_helpers([0,1,0,0,0]) == 4
puts count_zeros_with_helpers([0]) == 1
